internal/pkg/service/common/etcdop/op/util_test.go

```diff
@@ -0,0 +1,106 @@
+package op
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRemoveOverlaps(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name    string
+		in, out []TrackedOp
+	}{
+		{
+			name: "empty",
+			in:   nil,
+			out:  nil,
+		},
+		{
+			name: "get op",
+			in: []TrackedOp{
+				{Type: GetOp, Count: 1, Key: []byte("key"), RangeEnd: nil},
+			},
+			out: []TrackedOp{
+				{Type: GetOp, Count: 1, Key: []byte("key"), RangeEnd: nil},
+			},
+		},
+		{
+			name: "put op",
+			in: []TrackedOp{
+				{Type: PutOp, Count: 1, Key: []byte("key"), RangeEnd: nil},
+			},
+			out: []TrackedOp{
+				{Type: PutOp, Count: 1, Key: []byte("key"), RangeEnd: nil},
+			},
+		},
+		{
+			name: "delete op",
+			in: []TrackedOp{
+				{Type: DeleteOp, Count: 1, Key: []byte("key"), RangeEnd: nil},
+			},
+			out: []TrackedOp{
+				{Type: DeleteOp, Count: 1, Key: []byte("key"), RangeEnd: nil},
+			},
+		},
+		{
+			name: "get prefix op",
+			in: []TrackedOp{
+				{Type: GetOp, Key: []byte("some/prefix/foo005"), RangeEnd: []byte("some/prefix0"), Count: 3},
+				{Type: GetOp, Key: []byte("some/prefix/foo007"), RangeEnd: []byte("some/prefix0"), Count: 1},
+				{Type: GetOp, Key: []byte("some/prefix/"), RangeEnd: []byte("some/prefix0"), Count: 7},
+				{Type: GetOp, Key: []byte("some/prefix/foo003"), RangeEnd: []byte("some/prefix0"), Count: 5},
+				{Type: GetOp, Key: []byte("some/prefix/foo003"), RangeEnd: []byte("some/prefix0"), Count: 5},
+			},
+			out: []TrackedOp{
+				{Type: GetOp, Key: []byte("some/prefix/"), RangeEnd: []byte("some/prefix0"), Count: 7},
+			},
```

Comment:
Snad je to jasne z unit testu, ... je jedno v akom poradi su operacie reportovane, duplicity sa odstrania.